packages/runtime-vapor/__tests__/componentProps.spec.ts

```diff
@@ -0,0 +1,399 @@
+// NOTE: This test is implemented based on the case of `runtime-core/__test__/componentProps.spec.ts`.
+
+import { defineComponent, watchEffect } from 'vue'
+
+import type { FunctionalComponent } from '../src/component'
+import { getCurrentInstance } from '../src/component'
+import { render } from '../src/render'
+import { template } from '../src/template'
+import { children, setText } from '../src/dom'
+
+let host: HTMLElement
+const initHost = () => {
+  host = document.createElement('div')
+  host.setAttribute('id', 'host')
+  document.body.appendChild(host)
+}
+beforeEach(() => initHost())
+afterEach(() => host.remove())
+
+describe('component props (vapor)', () => {
+  test('stateful', () => {
+    let props: any
+    // TODO: attrs
+
+    const Comp = defineComponent({
+      props: ['fooBar', 'barBaz'],
+      render() {
+        const instance = getCurrentInstance()!
+        props = instance.props
+      },
+    })
+
+    render(
+      Comp as any,
+      {
+        get fooBar() {
+          return 1
+        },
+      },
+      host,
+    )
+    expect(props.fooBar).toEqual(1)
+
+    // test passing kebab-case and resolving to camelCase
+    render(
+      Comp as any,
+      {
+        get ['foo-bar']() {
+          return 2
+        },
+      },
+      host,
+    )
+    expect(props.fooBar).toEqual(2)
+
+    // test updating kebab-case should not delete it (#955)
+    render(
+      Comp as any,
+      {
+        get ['foo-bar']() {
+          return 3
+        },
+        get barBaz() {
+          return 5
+        },
+      },
+      host,
+    )
+    expect(props.fooBar).toEqual(3)
+    expect(props.barBaz).toEqual(5)
+
+    render(Comp as any, {}, host)
+    expect(props.fooBar).toBeUndefined()
+    expect(props.barBaz).toBeUndefined()
+    // expect(props.qux).toEqual(5) // TODO: attrs
+  })
+
+  test('stateful with setup', () => {
+    // FIXME: is it necessary?
+  })
```

Comment:
is it necessary?

https://github.com/vuejs/core/blob/d276a4f3e914aaccc291f7b2513e5d978919d0f9/packages/runtime-core/__tests__/componentProps.spec.ts#L66

Reply:
Yes. I think it's meaningful.